python/lsst/sims/coordUtils/CameraUtils.py

```diff
@@ -586,23 +783,23 @@ def focalPlaneCoordsFromPupilCoords(xPupil, yPupil, camera=None):
     coordinate (both in millimeters)
     """
 
-    if not isinstance(xPupil, numpy.ndarray) or not isinstance(yPupil, numpy.ndarray):
-        raise RuntimeError("You must pass numpy arrays of xPupil and yPupil to " +
-                               "focalPlaneCoordsFromPupilCoords")
-
-    if len(xPupil) != len(yPupil):
-        raise RuntimeError("You specified %d xPupil and %d yPupil coordinates " % (len(xPupil), len(yPupil)) +
-                           "in focalPlaneCoordsFromPupilCoords")
+    are_arrays = _validate_inputs([xPupil, yPupil],
+                                  ['xPupil', 'yPupil'], 'focalPlaneCoordsFromPupilCoords')
 
     if camera is None:
         raise RuntimeError("You cannot calculate focal plane coordinates without specifying a camera")
 
-    xPix = []
-    yPix = []
-    for x, y in zip(xPupil, yPupil):
-        cp = camera.makeCameraPoint(afwGeom.Point2D(x, y), PUPIL)
-        fpPoint = camera.transform(cp, FOCAL_PLANE)
-        xPix.append(fpPoint.getPoint().getX())
-        yPix.append(fpPoint.getPoint().getY())
+    if are_arrays:
+        xPix = []
+        yPix = []
+        for x, y in zip(xPupil, yPupil):
+            cp = camera.makeCameraPoint(afwGeom.Point2D(x, y), PUPIL)
+            fpPoint = camera.transform(cp, FOCAL_PLANE).getPoint()
+            xPix.append(fpPoint.getX())
+            yPix.append(fpPoint.getY())
+
+        return np.array([xPix, yPix])
 
-    return numpy.array([xPix, yPix])
```

Comment:
I like how you added a comment above that this section is for "if not are_arrays". Could help here too.